Sources/X509/RDNAttribute.swift

```diff
@@ -594,8 +482,14 @@ extension String {
             self = printable
         case .utf8(let utf8):
             self = utf8
-        default:
-            return nil
+        case .any(let value):
+            if let printable = try? ASN1PrintableString(asn1Any: value) {
+                self = String(printable)
+            } else if let utf8 = try? ASN1UTF8String(asn1Any: value) {
+                self = String(utf8)
+            } else {
+                return nil
+            }
```

Comment:
That's rather inefficient as this will very likely parse twice before returning nil. We should instead parse the data once, and switch over the identifier to decided if it is a UTF8 or Printable String like we do in the initialiser. 

The other option would be that we always "normalise" the data and parse it into  `.utf8String`, `.printableString` or `.any` where `.any` can never contain bytes that would be a valid `.utf8String` or `.printableString`.
This allows us to remove the parsing logic here, but more importantly, it allows us to get rid of the custom Equatable/Hashable conformance and their related types which would be big reduction in code complexity. The only place where we currently don't do this is in `RelativeDistinguishedName.Attribute.Value.init(asn1Any:)`.